app/webpacker/components/Panel/pages/ApprovePictures/AvatarCard.jsx

```diff
@@ -0,0 +1,170 @@
+import React, { useMemo, useState } from 'react';
+import {
+  Button,
+  ButtonGroup,
+  Header,
+  Segment,
+  Image,
+  Divider,
+  CardContent,
+  CardHeader,
+  CardMeta,
+  CardDescription,
+  Grid,
+  Card, Checkbox,
+} from 'semantic-ui-react';
+import { editPersonUrl } from '../../../../lib/requests/routes.js.erb';
+import I18n from '../../../../lib/i18n';
+
+function Avatar({
+  title, actions, imageUrl,
+}) {
+  return (
+    <Segment textAlign="center" padded>
+      <Header as="h3" content={title} />
+      <Image
+        src={imageUrl}
+        size="medium"
+        centered
+        style={{
+          objectFit: 'contain',
+          height: '250px',
+          width: '250px',
+        }}
+      />
+      <Divider horizontal />
+      <ButtonGroup toggle>
+        {actions.map((action) => (
+          <Button
+            key={action.label}
+            onClick={action.onClick}
+            color={action.color}
+            basic={action.basic}
+            disabled={action.disabled}
+          >
+            {action.label}
+          </Button>
+        ))}
+      </ButtonGroup>
+    </Segment>
+  );
+}
+
+export default function AvatarCard({
+  user, onReject, onApprove, onDefer,
+}) {
+  const [activeAvatarIndex, setActiveAvatarIndex] = useState(0);
+  const [avatarState, setAvatarState] = useState('');
```

Comment:
This is essentially duplicating state, because e.g. whenever this is set to 'defer', `onDefer` is also called. So the parent knows `avatarState` and should be able to provide it.

It's not quite that simple because of the additional `chooseRejectionReason`. But that could be converted into its own (boolean) state I think, while `avatarState` is hoisted up.